src/lib/structures/expr.ml

```diff
@@ -2569,14 +2570,22 @@ module Purification = struct
         in_e, add_let let_v let_e lets
 
       | (Sy.Lit _ | Sy.Form _), _ ->
-        let fresh_sy = Sy.fresh_skolem ~is_var:true "Pur-F" in
-        mk_term fresh_sy [] t.ty , add_let fresh_sy t lets
+        let fresh_var =
+          match Sy.fresh_skolem ~is_var:true "Pur-F" with
+          | Sy.Var v -> v
+          | _ -> assert false
+        in
+        mk_term (Sy.Var fresh_var) [] t.ty , add_let fresh_var t lets
 
       | _ -> (* detect ITEs *)
         match t.xs with
         | [_;_;_] when is_ite t.f ->
-          let fresh_sy = Sy.fresh_skolem ~is_var:true "Pur-Ite" in
-          mk_term fresh_sy [] t.ty , add_let fresh_sy t lets
+          let fresh_var =
+            match Sy.fresh_skolem ~is_var:true "Pur-Ite" with
```

Comment:
I'll check. Good catch.

Reply:
The function `Symbols.fresh_skolem` isn't necessary anymore. I create a two functions `fresh_skolem_string` and `fresh_skolem_name`. It would be better to extract the module of identifiers from the `Symbols` module and create a function `fresk_skolem` in `Var` but it's a small detail.